src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -46,10 +46,10 @@ public void addFolder(SpoonFolder o) {
 	@Override
 	public List<SpoonFile> getAllFiles() {
 		List<SpoonFile> result = new ArrayList<>();
-
+		Set<SpoonFile> distinctSet = new HashSet<>();
 		for (SpoonFile f : getFiles()) {
 			// we take care not to add a file that was already found in a folder
-			if (!result.contains(f)) {

```

Comment:
are your sure hashCode/equals of SpoonFile are correct? I'm not sure they are well tested.